src/calculation/CompositeReportBuilder.ts

```diff
@@ -0,0 +1,209 @@
+import { PopulationGroupResult, CalculationOptions, ExecutionResult } from '../types/Calculator';
+import { getCompositeScoringFromMeasure } from '../helpers/MeasureBundleHelpers';
+import { CompositeScoreType, PopulationType } from '../types/Enums';
+import { v4 as uuidv4 } from 'uuid';
+import { AbstractMeasureReportBuilder } from './AbstractMeasureReportBuilder';
+
+export type CompositeMeasureReport = fhir4.MeasureReport & {
+  group: [
+    fhir4.MeasureReportGroup & {
+      population: [
+        fhir4.MeasureReportGroupPopulation & { count: number },
+        fhir4.MeasureReportGroupPopulation & { count: number }
+      ];
+    }
+  ];
+};
+
+export class CompositeReportBuilder<T extends PopulationGroupResult> extends AbstractMeasureReportBuilder<
+  T,
+  CompositeMeasureReport
+> {
+  report: CompositeMeasureReport;
+  compositeScoringType: CompositeScoreType;
+  compositeFraction: { numerator: number; denominator: number };
+
+  constructor(compositeMeasure: fhir4.Measure, options: CalculationOptions) {
+    super(compositeMeasure, options);
+
+    this.compositeScoringType = getCompositeScoringFromMeasure(compositeMeasure) ?? 'all-or-nothing';
+    this.compositeFraction = { numerator: 0, denominator: 0 };
```

Comment:
Related to the one below although even less important because it is an object rather than a tuple, but thoughts on making this `{ denominator: 0, numerator: 0}` just to reinforce that pattern?

Reply:
I'm going to say no to this one since order never matters in an object